rhp/v4/rhp.go

```diff
@@ -0,0 +1,640 @@
+		ChallengeSignature types.Signature        `json:"challengeSignature"`
+	}
+	// RPCRenewContractResponse implements Object.
+	RPCRenewContractResponse struct {
+		HostInputs []types.V2SiacoinInput `json:"hostInputs"`
+	}
+	// RPCRenewContractSecondResponse implements Object.
+	RPCRenewContractSecondResponse struct {
+		RenterRenewalSignature  types.Signature         `json:"renterRenewalSignature"`
+		RenterSatisfiedPolicies []types.SatisfiedPolicy `json:"renterSatisfiedPolicies"`
+	}
+	// RPCRenewContractThirdResponse implements Object.
+	RPCRenewContractThirdResponse struct {
+		Basis          types.ChainIndex      `json:"basis"`
+		TransactionSet []types.V2Transaction `json:"transactionSet"`
+	}
+
+	// RPCModifySectorsRequest implements Request.
+	RPCModifySectorsRequest struct {
+		ContractID types.FileContractID `json:"contractID"`
+		Prices     HostPrices           `json:"prices"`
+		Actions    []WriteAction        `json:"actions"`
+		// A ChallengeSignature proves the renter can modify the contract.
+		ChallengeSignature types.Signature `json:"challengeSignature"`
+	}
+	// RPCModifySectorsResponse implements Object.
+	RPCModifySectorsResponse struct {
+		Proof []types.Hash256 `json:"proof"`
+	}
+	// RPCModifySectorsSecondResponse implements Object.
+	RPCModifySectorsSecondResponse struct {
+		RenterSignature types.Signature `json:"renterSignature"`
+	}
+	// RPCModifySectorsThirdResponse implements Object.
+	RPCModifySectorsThirdResponse struct {
+		HostSignature types.Signature `json:"hostSignature"`
+	}
+
+	// RPCLatestRevisionRequest implements Request.
+	RPCLatestRevisionRequest struct {
+		ContractID types.FileContractID `json:"contractID"`
+	}
+	// RPCLatestRevisionResponse implements Object.
+	RPCLatestRevisionResponse struct {
+		Contract types.V2FileContract `json:"contract"`
+	}
+
+	// RPCReadSectorRequest implements Request.
+	RPCReadSectorRequest struct {
+		Prices HostPrices    `json:"prices"`
+		Token  AccountToken  `json:"token"`
+		Root   types.Hash256 `json:"root"`
+		Offset uint64        `json:"offset"`
+		Length uint64        `json:"length"`
+	}
+	// RPCReadSectorResponse implements Object.
+	RPCReadSectorResponse struct {
+		Proof  []types.Hash256 `json:"proof"`
+		Sector []byte          `json:"sector"`
+	}
+
+	// RPCReadSectorStreamedResponse implements Object.
+	RPCReadSectorStreamedResponse struct {
+		Proof      []types.Hash256 `json:"proof"`
+		DataLength uint64          `json:"dataLength"`
+	}
+
+	// RPCWriteSectorStreamingRequest implements Request.
+	RPCWriteSectorStreamingRequest struct {
+		Prices     HostPrices   `json:"prices"`
+		Token      AccountToken `json:"token"`
+		Duration   uint64       `json:"duration"`
+		DataLength uint64       `json:"dataLength"` // extended to SectorSize by host
+	}
+
+	// RPCWriteSectorRequest implements Request.
+	RPCWriteSectorRequest struct {
+		Prices   HostPrices   `json:"prices"`
+		Token    AccountToken `json:"token"`
+		Duration uint64       `json:"duration"`
```

Comment:
what is this field for again?

Reply:
Sectors are written to disk without being associated with a contract. This gives the renter control over when the "temporary" sector expires